src/test/util/random.h

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2023 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_TEST_UTIL_RANDOM_H
+#define BITCOIN_TEST_UTIL_RANDOM_H
+
+#include <consensus/amount.h>
+#include <random.h>
+#include <test/util/setup_common.h>
+#include <uint256.h>
+
+#include <cstdint>
+
+static inline uint32_t InsecureRand32()
+{
+    return g_insecure_rand_ctx.rand32();
+}
+
+static inline uint256 InsecureRand256()
+{
+    return g_insecure_rand_ctx.rand256();
+}
+
+static inline uint64_t InsecureRandBits(int bits)
+{
+    return g_insecure_rand_ctx.randbits(bits);
+}
+
+static inline uint64_t InsecureRandRange(uint64_t range)
+{
+    return g_insecure_rand_ctx.randrange(range);
+}
+
+static inline bool InsecureRandBool()
+{
+    return g_insecure_rand_ctx.randbool();
+}
+
+/**
+ * Generate a semi-random CAmount up to COIN (a typical value) or MAX_MONEY (the
+ * full money range), with occasional edge case values of 0, 1, and MAX_MONEY.
+ */
+static inline CAmount SemiRandMoneyAmount()
```

Comment:
Not sure if we should imitate a fuzz engine in the unit tests to  pick special values. A fuzz engine is generally a lot better at finding those.

So I'd say, if you see value in testing more edge cases, I'd recommend to write a fuzz test, not a unit test with a hand written "edge-case finder".

Reply:
I thought about that, but unlike probing for a crash as fuzzers do, our unit tests are using the rand generators to check returned values, i.e. non-crash behavior, from those inputs and this change follows and improves on their intention.

We could simplify the helper to just `InsecureRandRange(MAX_MONEY)` if reviewers prefer, but I think the proposed version provides more robust and useful coverage.

Reply:
> unlike probing for a crash as fuzzers do, our unit tests are using the rand generators to check returned values

Conceptually unit tests and fuzz tests are no different, if you take away the fuzz engine. A fuzz test doing a check is no different than a unit test doing a check. There may be an implementation detail in the exit code (and other stuff) if one calls assert vs BOOST_REQUIRE or BOOST_CHECK, but in both cases the test exe will end unsuccessfully.

> I think the proposed version provides more robust and useful coverage.

Would be good to provide an example where this leads to increased line or branch coverage

Reply:
> We could simplify the helper to just `InsecureRandRange(MAX_MONEY)` if reviewers prefer

Happy to update with a concrete suggestion.

Reply:
> Would be good to provide an example where this leads to increased line or branch coverage

I think this is the key point. If we're going to write special randomness/edge case functions, we should be able to show they are improving test coverage. Otherwise have a fuzzer pick random values.

Reply:
Removed the edge cases. It uses the money range rather than only uint32, while remaining in the same vein as the existing unit test rand helpers.